src/hir/mod.rs

```diff
@@ -11,9 +12,32 @@ pub use hir::*;
 ///
 /// This function leverages `crate::syntax::parse` and `translator::Translator::translate`
 /// to hide away most of the complexity of `bulloak`'s internal compiler.
-pub fn translate(tree: &str) -> anyhow::Result<Hir> {
-    let ast = crate::syntax::parse(tree)?;
+pub fn translate(text: &str) -> anyhow::Result<Hir> {
```

Comment:
So the idea is that this function doesn't have to change at all.

```rust
fn entrypoint() -> anyhow::Result<Hir> {
	// This is the &str read from the file.
	let trees = "TREES";
	let hirs = trees.split(TREE_SEPARATOR).map(|t| translate(t)?);
	Combiner::new().combine(&hirs)	
}

fn translate(text: &str) -> anyhow::Result<Hir> {
	...
}
```

The above is illustrative, so it needs some tweaks, but that is the overall idea.

Reply:
I think we can start by just reporting the first error, and we can handle collecting all the errors later so that you don't have to deal with it right now.